be/src/storage/lake/compaction_task.h

```diff
@@ -28,24 +29,15 @@ class Rowset;
 
 class CompactionTask {
 public:
-    class Progress {
-    public:
-        int value() const { return _value.load(std::memory_order_acquire); }
-
-        void update(int value) { _value.store(value, std::memory_order_release); }
-
-    private:
-        std::atomic<int> _value{0};
-    };
-
     // CancelFunc is a function that used to tell the compaction task whether the task
     // should be cancelled.
     using CancelFunc = std::function<bool()>;
 
-    explicit CompactionTask(int64_t txn_id, VersionedTablet tablet, std::vector<std::shared_ptr<Rowset>> input_rowsets);
+    explicit CompactionTask(int64_t txn_id, VersionedTablet tablet, std::vector<std::shared_ptr<Rowset>> input_rowsets,
+                            CompactionTaskContext& context);
```

Comment:
```suggestion
    explicit CompactionTask(VersionedTablet tablet, std::vector<std::shared_ptr<Rowset>> input_rowsets,
                            CompactionTaskContext* context);
```
`CompactionTaskContext` contains `txn_id` already.

Reply:
Should also remove the member field `_txn_id`.